pkg/schema/service.go

```diff
@@ -0,0 +1,35 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package schema
+
+//go:generate mockgen -destination=mock/schema_service.go -package=mock -mock_names=Service=Service . Service
+
+import (
+	"context"
+
+	commschema "github.com/conduitio/conduit-commons/schema"
+	"github.com/conduitio/conduit/pkg/foundation/cerrors"
+)
+
+var (
+	ErrSchemaNotFound = cerrors.New("schema not found")
+)
+
+type Service interface {
```

Comment:
Not for this PR, but I'll mention it before I forget. I realized that having a builtin schema registry could create the expectation that the [`avro.decode`](https://conduit.io/docs/processors/builtin/avro.decode) and [`avro.encode`](https://conduit.io/docs/processors/builtin/avro.encode) processors would be able to use the internal schema registry. It would be a bit strange if we had to configure Conduit to point to a schema registry and then again supply the URL to the processors separately. So ideally the processors should use this service to create and get schemas. Not sure if this was included in the design or not.

Reply:
I created an issue for this so we can discuss it there: https://github.com/ConduitIO/conduit/issues/1653.